src/core.ts

```diff
@@ -161,11 +148,7 @@ export function createStarport<T extends Component>(
         {
           ref: el,
           id,
-          style: {
-            transitionProperty: 'all',
-            transitionDuration: `${context.options.duration}ms`,
-            transitionTimingFunction: context.options.easing,
-          },
+          style: defaultStyle(context),
```

Comment:
I don't like this refactoring as it makes the code harder to read and maintain (`defaultStyle` have too much logic inside and not clear what it is outputting)